File: swift/data_structure/base/1406.swift
var left: [Character] = Array(readLine()!)
var right: [Character] = []

for _ in 0..<Int(readLine()!)! {
    let command = readLine()!
    switch command {
        case "L":
            if let c = left.popLast() {
                right.append(c)
            }
            break
        case "D": 
            if let c = right.popLast() {
                left.append(c)
            }
            break
        case "B":
            if !left.isEmpty { left.removeLast() }
            break
        case let p where p.hasPrefix("P "):
            left.append(p.dropFirst(2).first!)
            break
        default:
            break
        // 아래 코드로 진행할 경우 0.3초 시간 초과
        // split을 사용하면 문자열이 길 경우 비효율적이므로 dropFirst 후 first 사용하는 게 효율적
        // default:
        //     left.append(command.split(separator: " ")[1].first!)
        
    }
}

print(String(left + right.reversed()))

// 에디터

File: javascript/data_structure/base/1406.js
// let [s, n, ...input] = require('fs').readFileSync('input.txt').toString().split('\n');
let [s, n, ...input] = require('fs').readFileSync('/dev/stdin').toString().split('\n');
let left = s.split("");
let right = [];
    
for (let i = 0; i < n; i++) {
    const c = input[i];
    const command = c[0];
    
    switch (command) {
        case "L":
            if (left.length > 0) { right.push(left.pop()); }
            break
        case "D":
            if (right.length > 0) { left.push(right.pop()); }
            break
        case "B":
            if (left.length > 0) { left.pop(); }
            break
        case "P":
            left.push(c[2]);
            break
    }
}

console.log(left.join("") + right.reverse().join(""));